Implement the `~` merge operation (#234)
teemtee/fmf: docs/features.rst

string. For dictionaries it's possible to provide list of keys
which should be removed.

Substitution of current values can be done by appending a ``~`` suffix
to the key name. The pattern and replacement parameters need to be provided
as values in the form of ``<d>PATTERN<d>REPLACEMENT<d>``, where ``<d>`` is delimiter
which can be any character however such character cannot be then used withing PATTERN

> and REPLACEMENT text as escaping isn't supported. These input can be either a string or

Reviewer: typo: `These input` ...

Author: fixed 25275a1

list of strings.

The `re.sub`__ is used to do the substitution thus all feaures of ``re.Pattern`` can
be used (named groups, backreferencing...).

In the fmf file it is better to use single quotes ``'`` as they do not need such intensive escaping::

>     require~: ';^foo;foo-ng;'

Reviewer 2: Just a quick first-sight brainstorm: Instead of describing the delimiters and parsing them and escaping them wouldn't it be easier to have two keys?

```yaml
require~:
    pattern: foo
    replace: foo-ng
```

Or do you find to too long?

Author: @psss Some details in https://github.com/teemtee/tmt/discussions/2430#discussioncomment-9131018 

I think that fmf prefers shorter way of syntax so "sed"-like seems (IMO) to be a better fit.

Reviewer 3: It could be confusing because no other operation sets dict keys. I am thinking of how it fits along with the `+` operator where the value is the operand. It's hard to describe

Reviewer 2: Thanks for the pointer. I've read he discussion. Makes sense to keep the syntax concise.

string. For dictionaries it's possible to provide list of keys
which should be removed.

Substitution of current values can be done by appending a ``~``
suffix to the key name. The pattern and replacement parameters
need to be provided as values in the form of
``<d>PATTERN<d>REPLACEMENT<d>``, where ``<d>`` is delimiter which
can be any character however such character cannot be then used
within PATTERN and REPLACEMENT text as escaping isn't supported.
This input can be either a string or list of strings.

The `re.sub`__ is used to do the substitution thus all features of
``re.Pattern`` can be used (named groups, back referencing...).

In the fmf file it is better to use single quotes ``'`` as they do
not need such intensive escaping::

    require~: ';^foo;foo-ng;'
    recommend~:
      - '/python2-/python3-/'

__ https://docs.python.org/3/library/re.html#re.sub

Remove parent value only if it matches regular expression is done
using  the ``-~`` suffix. If value matches any of provided

> `regular expressions`__ it is removed. In the case of parent value
> being a list the item is removed from this list. If the parent value
> is a string value is set to the empty string. If parent value is
> a dictionary, the key is removed. These regular expressions can

Reviewer 3: ```suggestion
`regular expressions`__ it is removed. If the parent value
is a list, the matching item is removed from this list. If the parent value
is a string, the value is set to an empty string. If the parent value is
a dictionary, the matching key is removed. These regular expressions can
```

Small tweaks.

I just noticed that basically the `-` operator for string does the same thing :thinking:
